ada_2_fhir-r4/fhir/2_fhir_fhir_include.xsl

```diff
@@ -108,20 +108,22 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
         </xsl:if>
         
         <!-- if-statement to allow for local variables -->
-        <xsl:if test="$in//patient[not(patient)][not(@datatype = 'reference')]">
-            <xsl:variable name="identifier" select="(identificatienummer[@root = $oidBurgerservicenummer],identificatienummer[not(@root = $oidBurgerservicenummer)])[1]"/>
+        <xsl:variable name="patient" select="$in[local-name() = 'patient']"/>
```

Comment:
Ik weet niet hoe wenselijk het is om 2_fhir_fhir_include aan te passen speciaal voor onze use-case. De 'oude' manier (alleen buildFhirMetadata callen op een ada-instance, het lijkt mij dat die nog steeds mogelijk moet blijven) is nu kapot.

Sowieso kan dit na mijn laatste commit weer `$in/patient` worden, maar ook weer `$in//patient[not(patient)][not(@datatype = 'reference')]` volgens mij?

Reply:
Ben zo vrij geweest het bij te werken. Lijkt in alle gevallen weer te kloppen

Reply:
Dit was eigenlijk een bugfix. Dit bestand is breder bedoeld dan alleen het zib2020-project, maar zoekt wel instances adhv. de oid voor dit project (ten minste, dat is mijn aanname). Door de template een "plat" lijstje te voeren van ADA-instances te voeden krijg je ook een cleaner geheel.

Reply:
> De 'oude' manier (alleen buildFhirMetadata callen op een ada-instance, het lijkt mij dat die nog steeds mogelijk moet blijven) is nu kapot

Dus dat was waarom die identifier in patient en zorgverlener werkten terwijl het eigenlijk niet zou moeten werken ...
Waarom is dit nu kapot? `$in` is een lijst van ADA-instances, dat kunnen er 0, 1 of meer dan 1 zijn.

Reply:
> Dus dat was waarom die identifier in patient en zorgverlener werkten terwijl het eigenlijk niet zou moeten werken
Ik weet niet waar dit over gaat.

$in was geen lijst van ADA-instances. Wat $in is, is op zich redelijk ongedefineerd, maar 1 mechanisme om referenties te vullen is het embedden van bijv. een zorgaanbieder instance in een zorgverlener instance (dat gebeurt dan op informatiestandaardniveau). Vandaar dat de '//' ergens wel nodig is lijkt me. Dat dit nu aanslaat op de zib2020 oid is een vervanging van zoeken op local-name() en ik heb ergens weleens gemeld dat dit waarschijnlijk niet houdbaar is.

Reply:
Fair enough. Maar als je stelt dat het op informatiestandaard-niveau bepaald wordt wanneer dingen ge-embed worden, dan denk ik eigenlijk dat mijn wijziging wel de juiste is. Dan is het namelijk per informatiestandaard (of ADA-project) bepaald hoe instances in elkaar zitten en is het de driver op projectniveau die dingen uit elkaar moet trekken. Of je moet kiezen voor een generiek mechanisme dat de hele boel recursief doorzoekt en garandeert dat je alle ADA-instances kan vinden.

N,B. is er iets mis met zoeken op `local-name()`?

Reply:
Daar heb je een punt. Ik vind het vooral geen goede ontwikkeling dat we het van een behoorlijk elegante constructie nu weer complexer aan het maken zijn. We zijn in 2 stappen (stap 1 was het verwijderen van de _registratie template) weggetrokken van de situatie waar je 'gewoon' een XSLT over een ADA-instance kon gooien om een resultaat te krijgen. Nu moet je een altijd een driver hebben en vanaf deze wijziging een driver die nog relatief complex wordt ook. Ik weet niet of dit de acceptatie ten goede komt.

>N,B. is er iets mis met zoeken op local-name()?
Het gebruik van local-name() is m.i. symptoombestrijding van het gebrek aan namespace in ADA. Stel dat FHIR een f:patient element zou hebben, zou een template daar ook op aanslaan, terwijl we dat niet willen, we willen alleen een ADA patient hebben.

Reply:
Ik vind het niet zo gek als je het vergelijkt met een willekeurige softwarelibrary; die specificeert ook gewoon een interface waar je aan moet voldoen en in je eigen applicatie zorg je ervoor dat je jouw data daarmee overweg kunnen. Maar als je een betrouwbare template kan maken waar je ook geneste instances naar kan leveren en daarmee een stap kan overslaan, waarom niet?

Reply:
Ik heb even wat MP-voorbeelden bekeken hoe dat zit met conceptIds, en dat heeft inderdaad geen relatie meer met de zib2020-oid. Dus dan zou je de decor definitie in moeten duiken om die inheritance uit te zoeken etc.etc. Dus is 'betrouwbaar' mogelijk op het moment? Nee ik zou zeggen van niet. 

Ik draai e.e.a. terug aan mijn commits zo.